ocaml/xapi/xapi_vusb.ml

```diff
@@ -34,6 +34,14 @@ let create ~__context ~vM ~uSB_group ~other_config =
         another vusb can not create with the same USB_group. *)
     if vusbs <> [] then
       raise (Api_errors.Server_error(Api_errors.usb_group_conflict, [Ref.string_of uSB_group]));
+
+    (* All USBs that will be attached to the VM must be on the same host, as USB from host A can
+      not attach to the vm resident on host B. *)
+    let possible_hosts = Xapi_vm.get_possible_hosts ~__context ~vm:vM in
+    let pusb = Helpers.get_first_pusb ~__context uSB_group in
+    let host = Db.PUSB.get_host ~__context ~self:pusb in
+    if not (List.mem host possible_hosts) then
+      raise (Api_errors.Server_error (Api_errors.pusb_not_in_possible_hosts, [Ref.string_of pusb; String.concat ";" (List.map Ref.string_of possible_hosts)]));
```

Comment:
The error message is showing the references but I think it should show the UUIDs because this is how the user thinks about this.

Reply:
you mean pusb uuid host uuid, right? In the error message it is showing that.
```The VM can't boot on the host that belongs to the pusb; it can boot only on the following hosts.
pusb: 8b0ae24d-58b3-263d-6aee-68737629e015
```

Reply:
``` raise (Api_errors.Server_error (Api_errors.pusb_not_in_possible_hosts, [Db.PUSB.get_uuid ~__context ~self:pusb; 
        String.concat ";" (List.map (fun host -> Db.Host.get_uuid ~__context ~self:host) possible_hosts)]));
``` 
This can get uuid. Seems a bit long.

Reply:
```
let pusb_uuid  = Db.PUSB.get_uuid ~__context ~self:pusb in
let host_uuids = List.map (fun self -> Db.Host.get_uuid ~__context ~self) possible_hosts in
  raise Api_errors.(Server_error (pusb_not_in_possible_hosts, [ pusb_uuid;  host_uuids]))
```